server_windows.go

```diff
@@ -37,3 +44,76 @@ func (s *Server) toLocalPath(p string) string {
 
 	return lp
 }
+
+var kernel32, _ = syscall.LoadLibrary("kernel32.dll")
+var getLogicalDrivesHandle, _ = syscall.GetProcAddress(kernel32, "GetLogicalDrives")
+
+func bitsToDrives(bitmap uint32) []string {
+	var drive rune = 'A'
+	var drives []string
+
+	for bitmap != 0 {
+		if bitmap&1 == 1 {
+			drives = append(drives, string(drive))
+		}
+		drive++
+		bitmap >>= 1
+	}
+
+	return drives
+}
+
+func getDrives() ([]string, error) {
+	if ret, _, callErr := syscall.Syscall(uintptr(getLogicalDrivesHandle), 0, 0, 0, 0); callErr != 0 {
+		return nil, fmt.Errorf("GetLogicalDrives: %w", callErr)
+	} else {
+		drives := bitsToDrives(uint32(ret))
+		return drives, nil
+	}
+}
+
+type dummyDriveStat struct {
+	name string
+}
+
+func (s *dummyDriveStat) Name() string {
+	return s.name
+}
+func (s *dummyDriveStat) Size() int64 {
+	return 1024
+}
+func (s *dummyDriveStat) Mode() os.FileMode {
+	return os.FileMode(0755)
+}
+func (s *dummyDriveStat) ModTime() time.Time {
+	return time.Now()
+}
+func (s *dummyDriveStat) IsDir() bool {
+	return true
+}
+func (s *dummyDriveStat) Sys() any {
+	return nil
+}
+
+type WinRoot struct {
+	dummyFile
+}
+
+func (f *WinRoot) Readdir(int) ([]os.FileInfo, error) {
```

Comment:
Be aware of this implementation detail from `os.File.Readdir`:
> If n > 0, Readdir returns at most n FileInfo structures. In this case, if Readdir returns an empty slice, it will return a non-nil error explaining why. At the end of a directory, the error is io.EOF.

This behavior has to be respected in our implementation.

As is, we would end up in an infinite loop of `Readdir` because we will never return the eventually expected `io.EOF`.

Reply:
Good point!

Hmm.. I see that currently, in server.go:
```
func (p *sshFxpReaddirPacket) respond(svr *Server) responsePacket {
	[..]
	dirents, err := f.Readdir(128)
	if err != nil {
		return statusFromError(p.ID, err)
	}
```

Does that mean the server is currently unable to list directories with > 128 entries?

Reply:
I think we are supposed to only return `EOF` once we returned an empty slice - So we'd have to implement keeping track of subsequent calls in some way.

Reply:
> So we'd have to implement keeping track of subsequent calls in some way.

Ok have done that - What about the 128 limit in `server.go`, though?

Reply:
> Does that mean the server is currently unable to list directories with > 128 entries?

No, it only returns 128 entries at a time. The client is to request a `SSH_FXP_READDIR` repeatedly until it receives an `EOF`: https://github.com/pkg/sftp/blob/v1.13.6/client.go#L363-L366